app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,49 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_to_submit, only: :new
+  before_action :authenticate_user!, only: :create
+  before_action :parent, only: :create
+
+  helper_method :new_comment, :parent, :comment
+
+  def new
+  end
+
+  def create
+    @comment = @parent.comments.build(comment_params)
+    if @comment.save
+      redirect_to @parent
+    elsif !@comment.save
+      flash[:inline] = @comment.errors.messages[:text].join(", ")
+      if @parent.kind_of?(Post)
+        redirect_to @parent
+      else
+        redirect_to comment_reply_comment_path(@parent)
+      end
```

Comment:
I honestly feel like this could be just `render :new` here. The text of the parent will be on that view any way so I'm not convinced this conditional is worth sending them back to the post page if the comment came was on a post. thoughts?